src/Symfony/Component/Validator/Tests/Constraints/UniqueValidatorTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\Unique;
+use Symfony\Component\Validator\Constraints\UniqueValidator;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+class UniqueValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator()
+    {
+        return new UniqueValidator();
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Validator\Exception\UnexpectedTypeException
+     */
+    public function testExpectsUniqueConstraintCompatibleType()
+    {
+        $this->validator->validate('', new Unique());
+    }
+
+    /**
+     * @dataProvider getValidValues
+     */
+    public function testValidValues($value)
+    {
+        $this->validator->validate($value, new Unique());
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidValues()
+    {
+        return array(
+            array(array(null)),
+            array(array(5)),
+            array(array('a')),
+            array(array()),
+            array(array(new \stdClass())),
+            array(array(true, false)),
+            array(array(1, 2, 3, 4, 5, 6)),
```

Comment:
To improve readability, you could use `yield` and give each provided data set a name:

```diff
- array(array(1, 2, 3, 4, 5, 6)),
+ yield 'unique integers' => array(1, 2, 3, 4, 5, 6);
```